File: Python/014. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        ans = ""
        n = len(strs)
        if n == 0:
            return ans
        i = 0
        tag = True
        while tag:
            if i < len(strs[0]):
                c = strs[0][i]
            else:
                break
            for j in range(1, n):
                if i < len(strs[j]) and c == strs[j][i]:
                    pass
                else:
                    tag = False
            if tag:
                ans = ans + c
            i += 1
        return ans

File: Go/014. Longest Common Prefix.go
func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	for i := 0; i < len(strs[0]); i++ {
		for k := 1; k < len(strs); k++ {
			if i >= len(strs[k]) || strs[k][i] != strs[0][i] {
				return strs[0][0 : i]
			}
		}
	}
	return strs[0]
}
